DQMOffline/Trigger/plugins/Tau3MuMonitor.cc

```diff
@@ -0,0 +1,153 @@
+  tau1DPhi_    = nullptr;
+  tau1DMass_   = nullptr;
+  tau2DEtaPhi_ = nullptr;
+}
+
+Tau3MuMonitor::~Tau3MuMonitor()
+{
+  if (genTriggerEventFlag_) delete genTriggerEventFlag_;
+}
+
+// shape the content of a "histogram PSet"
+void Tau3MuMonitor::fillHistoPSetDescription(edm::ParameterSetDescription & pset)
+{
+  pset.add<unsigned int>( "nbins" );
+  pset.add<double>      ( "xmin"  );
+  pset.add<double>      ( "xmax"  );
+}
+
+// read the information packed into an "histogram PSet"
+MEbinning Tau3MuMonitor::getHistoPSet(edm::ParameterSet pset)
+{
+  return MEbinning{
+    pset.getParameter<unsigned int>("nbins"),
+    pset.getParameter<double >     ("xmin" ),
+    pset.getParameter<double >     ("xmax" ),
+  };
+}
+
+// book histograms
+void Tau3MuMonitor::bookHistograms(DQMStore::IBooker     & ibooker,
+                                   edm::Run const        & iRun,
+                                   edm::EventSetup const & iSetup) 
+{  
+  std::string histname;
+
+  std::string currentFolder = folderName_ ;
+  ibooker.setCurrentFolder(currentFolder.c_str());
+  
+  // tau 3 mu 1D pt
+  histname = "tau1DPt"; 
+  tau1DPt_ = ibooker.book1D(histname, "", pt_binning_.nbins, pt_binning_.xmin, pt_binning_.xmax);
+  tau1DPt_->setAxisTitle("3-#mu p_{T} [GeV]", 1);
+  tau1DPt_->setAxisTitle("counts"           , 2);
+
+  // tau 3 mu 1D eta
+  histname  = "tau1DEta"; 
+  tau1DEta_ = ibooker.book1D(histname, "", eta_binning_.nbins, eta_binning_.xmin, eta_binning_.xmax);
+  tau1DEta_->setAxisTitle("3-#mu #eta", 1);
+  tau1DEta_->setAxisTitle("counts"    , 2);
+
+  // tau 3 mu 1D phi
+  histname  = "tau1DPhi"; 
+  tau1DPhi_ = ibooker.book1D(histname, "", phi_binning_.nbins, phi_binning_.xmin, phi_binning_.xmax);
+  tau1DPhi_->setAxisTitle("3-#mu #phi", 1);
+  tau1DPhi_->setAxisTitle("counts"    , 2);
+
+  // tau 3 mu 1D mass
+  histname   = "tau1DMass"; 
+  tau1DMass_ = ibooker.book1D(histname, "", mass_binning_.nbins, mass_binning_.xmin, mass_binning_.xmax);
+  tau1DMass_->setAxisTitle("mass_{3#mu} [GeV]", 1);
+  tau1DMass_->setAxisTitle("counts"           , 2);
+
+  // tau 3 mu 2D eta vs phi
+  histname  = "tau2DEtaPhi"; 
+  tau2DEtaPhi_ = ibooker.book2D(histname, "", eta_binning_.nbins, eta_binning_.xmin, eta_binning_.xmax,
+                                              phi_binning_.nbins, phi_binning_.xmin, phi_binning_.xmax);
+  tau2DEtaPhi_->setAxisTitle("3-#mu #eta", 1);
+  tau2DEtaPhi_->setAxisTitle("3-#mu #phi", 2);
+  
+  // Initialize the GenericTriggerEventFlag
+  if ( genTriggerEventFlag_ && genTriggerEventFlag_->on() ) genTriggerEventFlag_->initRun( iRun, iSetup );
+
+}
+
+void Tau3MuMonitor::analyze(edm::Event const& iEvent, edm::EventSetup const& iSetup)
+{
+
+  // require the trigger to be fired
+  if (genTriggerEventFlag_->on() && !genTriggerEventFlag_->accept(iEvent, iSetup) ) return;
+
+  // get ahold of the tau(3mu) collection
+  edm::Handle<reco::CompositeCandidateCollection> tauHandle;
+  iEvent.getByToken( tauToken_, tauHandle );
```

Comment:
can you please add a check of validity / availability of the Handle ?
because if it is not available, the code will simply crash ... and indeed I'm expecting the integration test to crash because of this ..... 
did you try 
`runTheMatrix -l limited`
?